File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.cpp
class Solution {
public:
    vector<int> getMaximumXor(vector<int>& nums, int maximumBit) {
        int n = nums.size();
        vector<int> arr(n);
        int XOR = 0;
            
        for(int j =0; j< n; j++) {
                XOR = XOR ^ nums[j];
        }
        
        int mask = (1<<maximumBit) - 1;
        
        for(int x=0; x<n; x++) {
            int k = XOR ^ mask;
            arr[x] = k;
            XOR = XOR ^ nums[n-x-1];
                
        }
        
        
        
        return arr;
    }
};

File: 1829-maximum-xor-for-each-query/1829-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        n = len(nums)
        arr = []
        XOR =0;
        
        for i in range(n):
            XOR ^= nums[i];
            
        mask = (1<<maximumBit) - 1
        
        for i in range(n):
            k = XOR ^ mask
            arr.append(k)
            XOR ^= nums[n-i-1]
            
        return arr